test/torchaudio_unittest/prototype/emformer_test_impl.py

```diff
@@ -0,0 +1,132 @@
+import torch
+from torchaudio_unittest.common_utils import TestBaseMixin, torch_script
+from torchaudio.prototype import Emformer
+
+
+class EmformerTestImpl(TestBaseMixin):
+    def _gen_model(self, input_dim, right_context_length):
+        emformer = Emformer(
+            input_dim,
+            8,
+            256,
+            3,
+            segment_length=4,
+            left_context_length=30,
+            right_context_length=right_context_length,
+            max_memory_size=1,
+        ).to(device=self.device, dtype=self.dtype)
+        return emformer
+
+    def _gen_inputs(self, input_dim, batch_size, num_frames, right_context_length):
+        input = torch.rand(batch_size, num_frames, input_dim).to(
+            device=self.device, dtype=self.dtype
+        )
+        lengths = torch.randint(1, num_frames - right_context_length, (batch_size,)).to(
+            device=self.device, dtype=self.dtype
+        )
```

Comment:
You have to add something like below to ensure that `max(lengths)` equals to `inpute.size(1) - right_context_length`.
```python3
lenghts[0] = num_frames - right_context_length
```